pkg/signing/signing.go

```diff
@@ -322,23 +324,27 @@ func Verify(cpath, keyServiceURI string, id uint32, isGroup bool, authToken stri
 		if err != nil {
 			// if theres a error, thats proboly becuse we dont have a local key
 
-			// download the key
-			sylog.Infof("Downloading key: %s...", fingerprint[24:])
-			netlist, err := sypgp.FetchPubkey(fingerprint, keyServiceURI, authToken, noPrompt)
-			if err != nil {
-				return fmt.Errorf("could not fetch public key from server: %s", err)
-			}
-			sylog.Verbosef("key retrieved successfully!")
-
-			block, _ := clearsign.Decode(data)
-			if block == nil {
-				return fmt.Errorf("failed to parse signature block")
-			}
-
-			// verify the container
-			signer, err = openpgp.CheckDetachedSignature(netlist, bytes.NewBuffer(block.Bytes), block.ArmoredSignature.Body)
-			if err != nil {
-				return fmt.Errorf("signature verification failed: %s", err)
+			if !localVerify {
+				// download the key
+				sylog.Infof("Downloading key: %s; not in keyring of trust.", fingerprint[24:])
+				netlist, err := sypgp.FetchPubkey(fingerprint, keyServiceURI, authToken, noPrompt)
+				if err != nil {
+					return fmt.Errorf("could not fetch public key from server: %s", err)
+				}
+				sylog.Verbosef("key retrieved successfully!")
+
+				block, _ := clearsign.Decode(data)
```

Comment:
the signature block here should be the same as on line 303, you don't need to do this again, do you?

Reply:
Yes, we do need this if the container is signed with multiple entities. But line 303 can be removed, and line 290 needs to stay.